doc/man7/EVP_KDF_SSHKDF.pod

```diff
@@ -0,0 +1,175 @@
+=pod
+
+=head1 NAME
+
+EVP_KDF_SSHKDF - The SSHKDF EVP_KDF implementation
+
+=head1 DESCRIPTION
+
+Support for computing the B<SSHKDF> KDF through the B<EVP_KDF> API.
+
+The EVP_KDF_SSHKDF algorithm implements the SSHKDF key derivation function.
+It is defined in RFC 4253, section 7.2 and is used by SSH to derive IVs,
+encryption keys and integrity keys.
+Five inputs are required to perform key derivation: The hashing function
+(for example SHA256), the Initial Key, the Exchange Hash, the Session ID,
+and the derivation key type.
+
+=head2 Numeric identity
+
+B<EVP_KDF_SSHKDF> is the numeric identity for this implementation; it
+can be used with the EVP_KDF_CTX_new_id() function.
+
+=head2 Supported controls
+
+The supported controls are:
+
+=over 4
+
+=item B<EVP_KDF_CTRL_SET_MD>
+
+=item B<EVP_KDF_CTRL_SET_KEY>
+
+These controls work as described in L<EVP_KDF_CTX(3)/CONTROLS>.
+
+=item B<EVP_KDF_CTRL_SET_SSHKDF_XCGHASH>
+
+=item B<EVP_KDF_CTRL_SET_SSHKDF_SESSION_ID>
+
+These controls expects two arguments: C<unsigned char *buffer>, C<size_t length>
+
+They set the respective values to the first B<lenght> bytes of the buffer
+B<buffer>. If a value is already set, the contents are replaced.
+
+EVP_KDF_ctrl_str() takes two type strings for these controls:
+
+=over 4
+
+=item "xcghash"
+
+=item "session_id"
+
+The value string is used as is.
+
+=item "hexxcghash"
+
+=item "hexsession_id"
+
+The value string is expected to be a hexadecimal number, which will be
+decoded before being passed on as the control value.
+
+=back
+
+=item B<EVP_KDF_CTRL_SET_SSHKDF_TYPE>
+
+This control expects one argument: C<int mode>
+
+Sets the type for the SSHHKDF operation. There are six supported types:
+
+=over 4
+
+=item EVP_KDF_SSHKDF_TYPE_ININITAL_IV_CLI_TO_SRV
+
+The Initial IV from client to server.
+A single char of value 65 (ASCII char 'A').
+
+=item EVP_KDF_SSHKDF_TYPE_ININITAL_IV_SRV_TO_CLI
+
+The Initial IV from server to client
+A single char of value 66 (ASCII char 'B').
+
+=item EVP_KDF_SSHKDF_TYPE_ENCRYPTION_KEY_CLI_TO_SRV
+
+The Encryption Key from client to server
+A single char of value 67 (ASCII char 'C').
+
+=item EVP_KDF_SSHKDF_TYPE_ENCRYPTION_KEY_SRV_TO_CLI
+
+The Encryption Key from server to client
+A single char of value 68 (ASCII char 'D').
+
+=item EVP_KDF_SSHKDF_TYPE_INTEGRITY_KEY_CLI_TO_SRV
+
+The Integrity Key from client to server
+A single char of value 69 (ASCII char 'E').
+
+=item EVP_KDF_SSHKDF_TYPE_INTEGRITY_KEY_SRV_TO_CLI
+
+The Integrity Key from client to server
+A single char of value 70 (ASCII char 'F').
+
+=back
+
+EVP_KDF_ctrl_str() type string: "type"
+
+The value is a string of length one character. The only valid values
+are the numerical values of the ASCII caracters: "A" (65) to "F" (70).
```

Comment:
I'd drop the references to ASCII here, and just say the letters "A" to "F". We are taking a string argument here so we have to be prepared to do any conversion internally.

Reply:
This is from the RFC:
```
 o  Initial IV client to server: HASH(K || H || "A" || session_id)
      (Here K is encoded as mpint and "A" as byte and session_id as raw
      data.  "A" means the single character A, ASCII 65).
```
And I think it is a lot more helpful to a user, so they know that we do not want characters in any random charset, but specificalli in the ASCII charset as demanded by the RFC, ie we want the byte values not just the chracters.
This will avoid mistakes and confusion on non-ASCII platforms.
Perhaps I can mention here that using the macros (and listing them) is the preferred option ?